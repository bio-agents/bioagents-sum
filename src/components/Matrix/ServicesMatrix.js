import React, { PureComponent } from 'react'
import { connect } from 'react-redux'
import * as R from 'ramda'
import { Panel } from 'react-bootstrap'
import MatrixCell from './MatrixCell'
import MatrixCellWithLink from './MatrixCellWithLink'
import { getServicesCounts } from '../../selectors/servicesSelector'
import { hyphenDelimitedToCamelCased, config } from '../../bioagentsSum/common/helperFunctions'
import { AbstractionCategory, ALL_SERVICES } from '../../constants/stringConstants'

class ServicesMatrix extends PureComponent {
  render () {
    const { servicesCounts } = this.props

    const panelHeader = (
      <div className='center-text'>
        <h4>{'Category selection matrix '}</h4>
      </div>
    )

    return (
      <Panel bsStyle='warning' header={panelHeader}>
        <div className='matrix'>
          <MatrixCellWithLink
            linkTo={`/${ALL_SERVICES}`}
            text='All services'
            numberOfServices={servicesCounts[hyphenDelimitedToCamelCased(ALL_SERVICES)]}
          />
          <MatrixCell text={AbstractionCategory[0]} />
          <MatrixCell text={AbstractionCategory[1]} />
          <MatrixCell text={AbstractionCategory[2]} />
          <MatrixCell text={AbstractionCategory[3]} />
          {config.rows.map(row =>
            <div key={row.name}>
              <MatrixCell text={row.name} />
              {R.take(4, row.cells).map((cell, cellIndex) => R.isEmpty(cell) || !cell.route
                ? <MatrixCell key={cellIndex} />
                : <MatrixCellWithLink
                  key={cell.route}
                  linkTo={`/${cell.route}`}
                  image={process.env.PUBLIC_URL + `/images/${cell.image}`}
                  numberOfServices={servicesCounts[hyphenDelimitedToCamelCased(cell.route)]}
                />
              )}
            </div>
          )}
        </div>
      </Panel>
    )
  }
}

export default ServicesMatrix = connect(state => ({
  servicesCounts: getServicesCounts(state),
}))(ServicesMatrix)
