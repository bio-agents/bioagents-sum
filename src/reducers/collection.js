import * as R from 'ramda'
import * as Actions from '../constants/actionTypes'
import { configCollection } from '../bioagentsSum/common/helperFunctions'

const initialState = {
  activeCollection: '',
  userEnteredCollection: false,
}

export const collection = (state = initialState, action) => {
  const { type, payload } = action

  switch (type) {
    case Actions.SET_COLLECTION:
      return R.evolve({
        activeCollection: R.always(payload.collection !== '' ? payload.collection : configCollection),
        userEnteredCollection: R.always(payload.collection !== '' ? payload.userEnteredCollection : false),
      })(state)

    default:
      return state
  }
}
