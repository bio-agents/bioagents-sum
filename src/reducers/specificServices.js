import * as R from 'ramda'
import * as Actions from '../constants/actionTypes'
import { pickData } from '../bioagentsSum/services/index'

const initialState = {
  count: 0,
  list: [],
  persistExpiresAt: '',
  serviceLoading: false,
  serviceLoaded: false,
  citationsLoading: false,
  citationsLoaded: false,
}

export const specificServicesWithName = (servicesName = '') =>
  (state = initialState, action) => {
    const { type, payload } = action

    if (payload && payload.name !== servicesName) return state

    switch (type) {
      case Actions.SERVICES_FETCH:
        return R.evolve({
          serviceLoading: R.T,
          serviceLoaded: R.F,
          citationsLoading: R.F,
          citationsLoaded: R.F,
        })(state)

      case Actions.SERVICES_FETCH_SUCCESS: {
        const { count, list } = payload.service
        let finalList = pickData(list)

        if (state.list.length !== 0) {
          const oldListByIds = R.groupBy(R.prop('id'), state.list)
          finalList = R.map(agent => R.mergeDeepLeft(agent, oldListByIds[agent.id] ? oldListByIds[agent.id][0] : {}), finalList)
        }

        return R.evolve({
          count: R.always(count),
          list: R.always(finalList),
          serviceLoading: R.F,
          serviceLoaded: R.T,
          citationsLoading: R.T,
          persistExpiresAt: R.always(Date.now() + 600000), // 86400000 ms == 24 hours
        })(state)
      }

      case Actions.SERVICES_FETCH_FAILURE: {
        return R.evolve({
          serviceLoading: R.F,
          serviceLoaded: R.F,
        })(state)
      }

      case Actions.CITATIONS_FETCH_SUCCESS: {
        const { count, list } = payload.service
        const pickedData = pickData(list)

        return R.evolve({
          count: R.always(count),
          list: R.always(pickedData),
          citationsLoading: R.F,
          citationsLoaded: R.T,
        })(state)
      }

      case Actions.CITATIONS_FETCH_FAILURE:
        return R.evolve({
          citationsLoading: R.F,
          citationsLoaded: R.F,
        })(state)

      case Actions.UPTIME_FETCH_SUCCESS: {
        const { count, list } = payload.service
        const pickedData = pickData(list)

        return R.evolve({
          count: R.always(count),
          list: R.always(pickedData),
        })(state)
      }

      case Actions.UPTIME_FETCH_FAILURE: {
        // todo
      }

      default:
        return state
    }
  }
