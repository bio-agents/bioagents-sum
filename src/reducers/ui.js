import * as R from 'ramda'
import * as Actions from '../constants/actionTypes'

const initialState = {
  showReportPage: false,
}

export const ui = (state = initialState, action) => {
  const { type } = action

  switch (type) {
    case Actions.REPORT_PAGE_CHANGE_VISIBILITY:
      return R.evolve({
        showReportPage: R.not,
      })(state)

    default:
      return state
  }
}
