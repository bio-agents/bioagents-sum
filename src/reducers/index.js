import { combineReducers } from 'redux'
import { routerReducer } from 'react-router-redux'
import { reducer as reduxFormReducer } from 'redux-form'
import { specificServicesWithName } from './specificServices'
import * as R from 'ramda'
import { ui } from './ui'
import { collection } from './collection'
import { getServicesNames } from '../bioagentsSum/common/helperFunctions'

const bioagentsSumApp = () => {
  const reducers = R.map(specificServicesWithName, getServicesNames)
  const servicesReducers = R.zipObj(getServicesNames, reducers)

  return combineReducers(R.merge(servicesReducers, {
    ui,
    collection,
    router: routerReducer,
    form: reduxFormReducer,
  }))
}

export default bioagentsSumApp
