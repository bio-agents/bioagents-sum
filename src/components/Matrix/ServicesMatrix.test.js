import React from 'react'
import renderer from 'react-test-renderer'
import { ConnectedRouter } from 'react-router-redux'
import { createStore } from 'redux'
import createHistory from 'history/createBrowserHistory'
import reducers from '../../reducers'
import { Provider } from 'react-redux'
import ServicesMatrix from './ServicesMatrix'

const store = createStore(reducers(), {})
const history = createHistory()

it('Renders ServicesMatrix correctly', () => {
  const component = renderer.create(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <ServicesMatrix />
      </ConnectedRouter>
    </Provider>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
