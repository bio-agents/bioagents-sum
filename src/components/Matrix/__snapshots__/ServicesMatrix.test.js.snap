exports[`test Renders ServicesMatrix correctly 1`] = `
<div
  className="panel panel-warning"
  id={undefined}>
  <div
    className="panel-heading">
    <div
      className="center-text panel-title">
      <h4>
        Category selection matrix 
      </h4>
    </div>
  </div>
  <div
    className="panel-body">
    <div
      className="matrix">
      <div
        className="square light-grey-background greyscale">
        <a
          href="/all-services"
          onClick={[Function]}>
          <div
            className="square-content">
            <div
              className="table">
              <div
                className="table-cell text-big">
                <span>
                  All services
                </span>
                <div
                  className="text-little">
                  0
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
      <div
        className="square dark-grey-background">
        <div
          className="square-content">
          <div
            className="table">
            <div
              className="table-cell text-big">
              <span>
                1D sequence
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="square dark-grey-background">
        <div
          className="square-content">
          <div
            className="table">
            <div
              className="table-cell text-big">
              <span>
                2D topology
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="square dark-grey-background">
        <div
          className="square-content">
          <div
            className="table">
            <div
              className="table-cell text-big">
              <span>
                3D structure
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="square dark-grey-background">
        <div
          className="square-content">
          <div
            className="table">
            <div
              className="table-cell text-big">
              <span>
                xD omics
              </span>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          className="square dark-grey-background">
          <div
            className="square-content">
            <div
              className="table">
              <div
                className="table-cell text-big">
                <span>
                  DNA
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="square light-grey-background greyscale">
          <a
            href="/dna-1d-services"
            onClick={[Function]}>
            <div
              className="square-content">
              <div
                className="table">
                <div
                  className="table-cell text-big">
                  <img
                    className="responsive-img grey"
                    src="/images/dna-1d-services.png" />
                  <div
                    className="text-little">
                    0
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="square light-grey-background greyscale">
          <a
            href="/dna-2d-services"
            onClick={[Function]}>
            <div
              className="square-content">
              <div
                className="table">
                <div
                  className="table-cell text-big">
                  <img
                    className="responsive-img grey"
                    src="/images/dna-2d-services.png" />
                  <div
                    className="text-little">
                    0
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="square light-grey-background greyscale">
          <a
            href="/dna-3d-services"
            onClick={[Function]}>
            <div
              className="square-content">
              <div
                className="table">
                <div
                  className="table-cell text-big">
                  <img
                    className="responsive-img grey"
                    src="/images/dna-3d-services.png" />
                  <div
                    className="text-little">
                    0
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="square light-grey-background greyscale">
          <a
            href="/dna-xd-services"
            onClick={[Function]}>
            <div
              className="square-content">
              <div
                className="table">
                <div
                  className="table-cell text-big">
                  <img
                    className="responsive-img grey"
                    src="/images/dna-xd-services.png" />
                  <div
                    className="text-little">
                    0
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
      <div>
        <div
          className="square dark-grey-background">
          <div
            className="square-content">
            <div
              className="table">
              <div
                className="table-cell text-big">
                <span>
                  RNA
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="square light-grey-background greyscale">
          <a
            href="/rna-1d-services"
            onClick={[Function]}>
            <div
              className="square-content">
              <div
                className="table">
                <div
                  className="table-cell text-big">
                  <img
                    className="responsive-img grey"
                    src="/images/rna-1d-services.png" />
                  <div
                    className="text-little">
                    0
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="square light-grey-background greyscale">
          <a
            href="/rna-2d-services"
            onClick={[Function]}>
            <div
              className="square-content">
              <div
                className="table">
                <div
                  className="table-cell text-big">
                  <img
                    className="responsive-img grey"
                    src="/images/rna-2d-services.png" />
                  <div
                    className="text-little">
                    0
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="square light-grey-background greyscale">
          <a
            href="/rna-3d-services"
            onClick={[Function]}>
            <div
              className="square-content">
              <div
                className="table">
                <div
                  className="table-cell text-big">
                  <img
                    className="responsive-img grey"
                    src="/images/rna-3d-services.png" />
                  <div
                    className="text-little">
                    0
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="square light-grey-background greyscale">
          <a
            href="/rna-xd-services"
            onClick={[Function]}>
            <div
              className="square-content">
              <div
                className="table">
                <div
                  className="table-cell text-big">
                  <img
                    className="responsive-img grey"
                    src="/images/rna-xd-services.png" />
                  <div
                    className="text-little">
                    0
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
      <div>
        <div
          className="square dark-grey-background">
          <div
            className="square-content">
            <div
              className="table">
              <div
                className="table-cell text-big">
                <span>
                  Protein
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="square light-grey-background greyscale">
          <a
            href="/protein-1d-services"
            onClick={[Function]}>
            <div
              className="square-content">
              <div
                className="table">
                <div
                  className="table-cell text-big">
                  <img
                    className="responsive-img grey"
                    src="/images/protein-1d-services.png" />
                  <div
                    className="text-little">
                    0
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="square light-grey-background greyscale">
          <a
            href="/protein-2d-services"
            onClick={[Function]}>
            <div
              className="square-content">
              <div
                className="table">
                <div
                  className="table-cell text-big">
                  <img
                    className="responsive-img grey"
                    src="/images/protein-2d-services.png" />
                  <div
                    className="text-little">
                    0
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="square light-grey-background greyscale">
          <a
            href="/protein-3d-services"
            onClick={[Function]}>
            <div
              className="square-content">
              <div
                className="table">
                <div
                  className="table-cell text-big">
                  <img
                    className="responsive-img grey"
                    src="/images/protein-3d-services.png" />
                  <div
                    className="text-little">
                    0
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="square light-grey-background greyscale">
          <a
            href="/protein-xd-services"
            onClick={[Function]}>
            <div
              className="square-content">
              <div
                className="table">
                <div
                  className="table-cell text-big">
                  <img
                    className="responsive-img grey"
                    src="/images/protein-xd-services.png" />
                  <div
                    className="text-little">
                    0
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
      <div>
        <div
          className="square dark-grey-background">
          <div
            className="square-content">
            <div
              className="table">
              <div
                className="table-cell text-big">
                <span>
                  Drugs and other small molecules
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="square light-grey-background greyscale">
          <a
            href="/drug-1d-services"
            onClick={[Function]}>
            <div
              className="square-content">
              <div
                className="table">
                <div
                  className="table-cell text-big">
                  <img
                    className="responsive-img grey"
                    src="/images/drug-1d-services.png" />
                  <div
                    className="text-little">
                    0
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="square light-grey-background greyscale">
          <a
            href="/drug-2d-services"
            onClick={[Function]}>
            <div
              className="square-content">
              <div
                className="table">
                <div
                  className="table-cell text-big">
                  <img
                    className="responsive-img grey"
                    src="/images/drug-2d-services.png" />
                  <div
                    className="text-little">
                    0
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="square light-grey-background greyscale">
          <a
            href="/drug-3d-services"
            onClick={[Function]}>
            <div
              className="square-content">
              <div
                className="table">
                <div
                  className="table-cell text-big">
                  <img
                    className="responsive-img grey"
                    src="/images/drug-3d-services.png" />
                  <div
                    className="text-little">
                    0
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="square light-grey-background greyscale">
          <a
            href="/drug-xd-services"
            onClick={[Function]}>
            <div
              className="square-content">
              <div
                className="table">
                <div
                  className="table-cell text-big">
                  <img
                    className="responsive-img grey"
                    src="/images/drug-xd-services.png" />
                  <div
                    className="text-little">
                    0
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
